sphinx_gallery/tests/test_gen_gallery.py

```diff
@@ -166,6 +168,23 @@ def test_config_backreferences(config_app):
     assert build_warn == ""
 
 
+def test_duplicate_files_warn(capsys, config_app):
+    """Test for a warning when two files with the same filename exist."""
+    files = ['./a/file1.py', './a/file2.py', 'a/file3.py', './b/file1.py']
+    msg = ("Duplicate file name(s) found. Having duplicate file names "
+           "will break some links. List of files: {}")
+    m = "['./b/file1.py']" if sys.version_info[0] >= 3 else "[u'./b/file1.py']"
+
+    # No warning because no overlapping names
+    check_duplicate_filenames(files[:-1])
+    warnings = config_app._warning.getvalue()
+    assert warnings == ''
+
+    # Warning because last file is named the same
+    check_duplicate_filenames(files)
+    warnings = config_app._warning.getvalue()
+    assert msg.format(m) in warnings
+
```

Comment:
two newlines before function def (assuming we have PEP8 checking on)

Reply:
We don't have pep8 checks and we probably should. For a project like sphinx-gallery it would make sense to fix all the PEP8 violations ans have a pep8 check in PRs. Maybe pytest-pep8?